server/storage/wal/repair.go

```diff
@@ -67,7 +67,7 @@ func Repair(lg *zap.Logger, dirpath string) bool {
 
 		case errors.Is(err, io.ErrUnexpectedEOF):
 			brokenName := f.Name() + ".broken"
-			bf, bferr := os.Create(brokenName)
+			bf, bferr := os.OpenFile(brokenName, os.O_WRONLY|os.O_CREATE, fileutil.PrivateFileMode)
```

Comment:
This removed `O_TRUNC` from the file open flags, is there any chance for file collision?  cc @ahrtr

Reply:
Right, let's add `os.O_TRUNC`.

We backup the whole wal file to a .broken file, so we should truncate the existing file if present (unlikely in practice).

Reply:
Added `os.O_TRUNC` back